guava/src/com/google/common/base/MoreObjects.java

```diff
@@ -39,21 +38,34 @@
 @GwtCompatible
 public final class MoreObjects {
   /**
-   * Returns the first of two given parameters that is not {@code null}, if either is, or otherwise
-   * throws a {@link NullPointerException}.
+   * Returns the first of the given parameters that is not {@code null}.
+   * If all are null, throws a {@link NullPointerException}.
    *
    * <p><b>Note:</b> if {@code first} is represented as an {@link Optional}, this can be
    * accomplished with {@link Optional#or(Object) first.or(second)}. That approach also allows for
    * lazy evaluation of the fallback instance, using {@link Optional#or(Supplier)
    * first.or(supplier)}.
    *
-   * @return {@code first} if it is non-null; otherwise {@code second} if it is non-null
-   * @throws NullPointerException if both {@code first} and {@code second} are null
+   * @return the first non-null parameter
+   * @throws NullPointerException if all parameters are null
    * @since 18.0 (since 3.0 as {@code Objects.firstNonNull()}).
    */
   @CheckReturnValue
   public static <T> T firstNonNull(@Nullable T first, @Nullable T second) {
-    return first != null ? first : checkNotNull(second);
+    return (first != null) ? first
+                           : checkNotNull(second);
+  }
+
+  @CheckReturnValue
+  public static <T> T firstNonNull(@Nullable T first, @Nullable T second, @Nullable T third, @Nullable T... rest) {
```

Comment:
I would add javadoc to this method as well, otherwise there is no documentation visible in IDEs at the _call site_.